src/pages/SchedulePage.js

```diff
@@ -1,119 +1,280 @@
+      });
+    }
+    let newCheckboxes = { ...checkboxes };
+    newCheckboxes[checkbox] = event.target.checked;
+    setCheckboxes(newCheckboxes);
+    localStorage.setItem(
+      `${year}-${semester}-checkbox-state`,
+      JSON.stringify(newCheckboxes)
+    );
+  };
 
-import * as actions from "../redux/modules/wishlist";
-
-function SchedulePage({ year, semester, wishlist, clearWishlist }) {
-  const _renderCourses = (dayNum, wishlist) =>
-    wishlist
-      .filter(
-        (course) =>
-          course.meetings.filter(
-            (meeting) => new Date(meeting.beginDate).getDay() === dayNum
-          ).length > 0
-      )
-      .map((course, i) => (
-        <div key={i} style={{ marginTop: "0.5em" }}>
-          <h3>{course.name}</h3>
-        </div>
-      ));
+  const _renderCourses = (dayNum) =>
+    schedule[dayToStr[dayNum]] !== undefined &&
+    Object.values(schedule[dayToStr[dayNum]]).map((course, i) => {
+      return <ScheduleCourse course={course} key={i} />;
+    });
 
   return (
-    <div style={{ padding: "2rem 10vw" }}>
-      <CourseCalendar>
-        <CalendarDay>
-          Monday
-          {_renderCourses(1, wishlist)}
-        </CalendarDay>
-        <CalendarDay>
-          Tuesday
-          {_renderCourses(2, wishlist)}
-        </CalendarDay>
-        <CalendarDay>
-          Wednesday
-          {_renderCourses(3, wishlist)}
-        </CalendarDay>
-        <CalendarDay>
-          Thursday
-          {_renderCourses(4, wishlist)}
-        </CalendarDay>
-        <CalendarDay>
-          Friday
-          {_renderCourses(5, wishlist)}
-        </CalendarDay>
-        <div>
-          <CalendarWeekend>
-            Saturday
-            {_renderCourses(6, wishlist)}
-          </CalendarWeekend>
-          <CalendarWeekend>
-            Sunday
-            {_renderCourses(0, wishlist)}
-          </CalendarWeekend>
-        </div>
-      </CourseCalendar>
-      {wishlist.length === 0 ? (
-        <span>No courses wishlisted yet!</span>
-      ) : (
-        <div>
-          <div
-            style={{ cursor: "pointer" }}
-            onClick={() => clearWishlist({ year, semester })}
-            onKeyPress={() => clearWishlist({ year, semester })}
-            role="button"
-            tabIndex={0}
-          >
-            Clear Wishlist
-          </div>
-          {}
-        </div>
-      )}
-    </div>
+    <Container>
+      <Calendar renderCourses={_renderCourses} />
+      <div
+        style={{
+          marginTop: "2rem",
+        }}
+      >
+        <Header>
+          <h2 className="wishlist">{`Wishlist (${wishlist.length})`}</h2>
+        </Header>
+        <WishlistCoursesList>
+          {wishlist.length === 0 ? (
+            <div
+              style={{
+                display: "flex",
+                alignItems: "center",
+                justifyContent: "center",
+                padding: "10px",
+              }}
+            >
+              Wishlist displayed here
```

Comment:
Make this div a styled component and also maybe change text to `"Your wishlist appears empty! Search for courses to add to your wishlist"` where `Search` or `Search for courses` is a link back to the homepage with the search bar.